must-gather/collection-scripts/gather_ceph_resources

```diff
@@ -245,18 +245,26 @@ for ns in $namespaces; do
 
         # Collecting snapshot info for ceph rbd volumes
         printf "collecting snapshot info for ceph rbd volumes \n" | tee -a  "${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
+        COMMAND_OUTPUT_FILE_VOL=${COMMAND_OUTPUT_DIR}/rbd_vol_info
+        COMMAND_OUTPUT_FILE_SNAP=${COMMAND_OUTPUT_DIR}/rbd_snap_info
         COMMAND_OUTPUT_FILE=${COMMAND_OUTPUT_DIR}/rbd_vol_and_snap_info
         # Inspecting ceph block pools for ceph rbd
         blockpools=$(timeout 60 oc get cephblockpools.ceph.rook.io -n openshift-storage -o jsonpath="{range .items[*]}{@.metadata.name}{'\n'}{end}")
         for bp in $blockpools; do
             images=$(timeout 60 oc -n openshift-storage exec "${HOSTNAME}"-helper -- bash -c "rbd ls -p $bp")
             for image in $images; do
-                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd info $image --pool $bp" >> "${COMMAND_OUTPUT_FILE}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-"${image}"-debug.log 2>&1 &
-                pids_ceph+=($!)
-                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd snap ls --all $image --pool $bp" >> "${COMMAND_OUTPUT_FILE}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-snap-"${image}"-debug.log 2>&1 &
+                printf "collecting vol and snapshot info for %s\n" "${image}" | tee -a  "${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
+                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd info $image --pool $bp" >> "${COMMAND_OUTPUT_FILE_VOL}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-image-"${image}"-debug.log 2>&1 &
+                { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd snap ls --all $image --pool $bp" >> "${COMMAND_OUTPUT_FILE_SNAP}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-rbd-snap-"${image}"-debug.log 2>&1 &
                 pids_ceph+=($!)
             done
         done
+        printf "copying vol and snapshot info for ceph rbd volumes \n" | tee -a  "${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
+        cat "${COMMAND_OUTPUT_FILE_VOL}" >> "${COMMAND_OUTPUT_FILE}"
+        cat "${COMMAND_OUTPUT_FILE_SNAP}" >> "${COMMAND_OUTPUT_FILE}"
+        printf "Deleting files \n" | tee -a  "${BASE_COLLECTION_PATH}"/gather-ceph-debug.log
+        rm -rf "${COMMAND_OUTPUT_FILE_VOL}"
+        rm -rf "${COMMAND_OUTPUT_FILE_SNAP}"
```

Comment:
Did you verify this by testing.?
There is no wait before you do this.
I think this task should be done after line 324.

Reply:
yes, it is working fine for me.

Reply:
@crombus is correct, you can't manipulate with the output without waiting for all involved commands to complete.  And it's not working for you either -- in the example you attached privately there are 95 `rbd image ...` lines (meaning that 95 `rbd info` outputs were captured) and 93 `SNAPID  NAME` lines (meaning 93 `rbd snap ls --all` outputs were captured)...

But there is an even bigger issue: the resulting `rbd_vol_and_snap_info` file would still be useless even if it didn't suffer from the above.  `rbd snap ls --all` output for a particular image should really follow `rbd info` output for the same image.  Otherwise, with all `rbd snap ls --all` outputs just dumped at the end as it is now, there is no way to determine which set of snapshots belongs to which image.

Reply:
@idryomov in that case, the only solution I see is to use wait and timeout the backend process as suggested by @crombus. I will very that it is working fine this way and makes changes to the PR accordingly.